login-app/src/components/Login.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+import { connect } from 'react-redux'
+import { login } from '../actions'
+
+class Login extends React.Component {
+    state = {
+        credentials: {
+            username: "",
+            password: ""
+        }
+    }
```

Comment:
Your stat only contains the username and password.  Consequently, I might just remove them from the credentials object and put them directly into state

Reply:
If you do that, you actually don't need the spread operator as setState only changes what was changed and leaves the remainder untouched which is different from functional component's useState.